pkg/keys/ed25519_test.go

```diff
@@ -37,6 +38,32 @@ func (Ed25519Suite) TestUnmarshalEd25519(c *C) {
 	c.Assert(verifier.UnmarshalPublicKey(badKey), IsNil)
 }
 
+func FuzzUnmarshalEd25519(f *testing.F) {
+	f.Fuzz(func(t *testing.T, s string) {
+		if len(s) <= 32 { // ed25519 requires strings of len greater than 32
+			t.Skip()
+		}
+		c := &C{}
+
+		pub, _, err := ed25519.GenerateKey(strings.NewReader(s))
+		c.Assert(err, IsNil)
+
+		publicKey, err := json.Marshal(map[string]string{
+			"public": hex.EncodeToString(pub),
```

Comment:
Sorry, I don't understand your comment. Could you please explain?

Reply:
Yeah, that was a little unclear, sorry. Backing up a step: what is the point of these fuzz tests?

We get nonsense input from the Go fuzzer in whatever format/type we want, and we call `UnmarshalPublicKey` with it to try to exercise weird edge cases. `UnmarshalPublicKey` can fail in a couple of ways:

1. Unmarshalling the value of `Value` into a `p256verifier`. I don't think this is ever going to fail in an interesting way, because that would be an error in `json.Decode`, which is already fuzzed extensively.
2. [`elliptic.Unmarshal(elliptic.P256(), p.PublicKey)`](https://pkg.go.dev/crypto/elliptic#Unmarshal)

So how do we feed our garbage fuzz data into `UnmarshalPublicKey` in a way that will make (2) do something interesting? I think what you're doing (making a `Value` with a legitimate JSON encoding of a `data.PublicKey`) is a great idea to exercise good code paths. But the value of `"public"` could be any `byte[]`! What should we put there? I can think of a few ideas:

1. Some bytes using `hex.EncodeToString`: this is great, because it winds up being in the expected format and possibly triggering crypto-related failures.
2. Arbitrary data: this might trigger *other* code paths

You're currently doing (1), which makes sense, as that's the format that's expected. But doing (2) as well would maybe uncover other situations we're interested in.

Hopefully that was clearer.